lib/mapper.js

```diff
@@ -314,6 +321,34 @@ function formatMetadata(props){
 }
 
 /**
+ * Flatten selectively based on your settings. You can either stringify, flatten, or drop the properties.
+ * Intercom rejects nested objects so you must choose a method.
+ *
+ * @param {Object} obj
+ * @param {Object} blacklisted
+ * @return {Object} ret
+ * @api private
+ */
+
+function formatNested(obj, blacklisted){
+  return foldl(function(attrs, value, key){
+    var trait = blacklisted[key];
+    var pair = {};
+    pair[key] = value;
+
+    if (trait === 'stringify') {
+      attrs[key] = JSON.stringify(value);
+    } else if (trait === 'drop') {
+      return attrs;
+    } else {
+      attrs = extend(flatten(pair), attrs);
```

Comment:
So then the default will still be to flatten? If they are concerned about column count, maybe they should be able to pick a default so they don't have to go back and change there settings every time a new field is added